src/request_items.rs

```diff
@@ -177,19 +177,21 @@ impl RequestItems {
             .any(|item| matches!(item, RequestItem::FormFile(..)))
     }
 
-    pub fn headers(&self) -> Result<(HeaderMap<HeaderValue>, Vec<HeaderName>)> {
+    pub fn headers(&self) -> Result<(HeaderMap<HeaderValue>, HashSet<HeaderName>)> {
         let mut headers = HeaderMap::new();
-        let mut headers_to_unset = vec![];
+        let mut headers_to_unset = HashSet::new();
```

Comment:
I wonder what would be the downside if I disable `#[deny(clippy::mutable_key_type)]` for this line?

Reply:
That should be ok. The lint's [documentation](https://rust-lang.github.io/rust-clippy/master/#mutable_key_type) specifically calls out the `bytes` crate as causing false positives, and I think that's what's causing `HeaderName` to be flagged as possibly having interior mutability. But there's no actual way to modify a `HeaderName` while it's inside the `HashSet`.